### PR TITLE
add jar-infer artifact

### PR DESCRIPTION
**Pull request by msridhar**
Artifact name is "com.uber.nullaway:jar-infer"

### CODE REVIEW

File: jar-infer/build.gradle [hunk 1]
```diff
@@ -26,16 +26,9 @@ dependencies {
     compile group: 'com.ibm.wala', name: 'com.ibm.wala.shrike', version: '1.4.3'
     compile group: 'com.ibm.wala', name: 'com.ibm.wala.util', version: '1.4.3'
     compile group: 'com.ibm.wala', name: 'com.ibm.wala.core', version: '1.4.3'
-    compile group: 'com.ibm.wala', name: 'com.ibm.wala.cast', version: '1.4.3'
```
Comment (msridhar) [inline jar-infer/build.gradle]: I also removed a bunch of unnecessary WALA dependencies

File: jar-infer/build.gradle [hunk 2]
```diff
@@ -26,16 +26,9 @@ dependencies {
     compile group: 'com.ibm.wala', name: 'com.ibm.wala.shrike', version: '1.4.3'
     compile group: 'com.ibm.wala', name: 'com.ibm.wala.util', version: '1.4.3'
```
Comment (kageiit) [inline jar-infer/build.gradle]: can we move these to deps.gradle as well?

**Comment by kageiit** [comment]
it should be "com.uber.nullaway:jar-infer" right?

File: jar-infer/build.gradle [hunk 3]
```diff
@@ -23,19 +23,9 @@ repositories {
...
-    compile group: 'com.ibm.wala', name: 'com.ibm.wala.cast.js.rhino', version: '1.4.3'
-    compile group: 'com.ibm.wala', name: 'com.ibm.wala.cast.java', version: '1.4.3'
-    compile group: 'com.ibm.wala', name: 'com.ibm.wala.cast.java.ecj', version: '1.4.3'
-    // uncomment to use wala.dalvik or wala.scandroid
-    //compile group: 'com.ibm.wala', name: 'com.ibm.wala.dalvik', version: '1.4.3'
-    //compile group: 'com.ibm.wala', name: 'com.ibm.wala.scandroid', version: '1.4.3'
-
+    compile deps.build.wala
     compile deps.build.guava
+    compile deps.build.commonsIO
```
Comment (lazaroclapp) [inline jar-infer/build.gradle]: Just curious, what is this?

Comment (msridhar) [reply jar-infer/build.gradle]: Previously the jar-infer code was depending on this but via a transitive dependence from some other WALA project; I made the dependence explicit.  You'll have to ask @subarnob what we use it for 😄

**Pull request merged**